### PR TITLE
Updates for beta 4.

### PR DESCRIPTION
**Pull request by mcdee**
Updates as per release v1.4.0-beta.4 of the Ethereum consensus specs.

### CODE REVIEW

File: spec/deneb/kzgcommitmentinclusionproof.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	return nil
+}
+
+// MarshalJSON implements json.Marshaler.
+func (k *KZGCommitmentInclusionProof) MarshalJSON() ([]byte, error) {
+	if k == nil {
+		return nil, errors.New("value nil")
+	}
+
+	return []byte(fmt.Sprintf(`["%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x","%#x"]`,
```
Comment (avalonche) [inline spec/deneb/kzgcommitmentinclusionproof.go:83]: can do a loop based on the proof depth here?
```
  proofs := make([]string, kzgCommitmentProofElements)
  for i, value := range *k {
      proofs[i] = fmt.Sprintf("%#x", value)
  }

  jsonStr := "[" + strings.Join(strElements, ",") + "]"
```

File: spec/deneb/blobsidecar_yaml.go [hunk 2]
```diff
@@ -18,33 +18,35 @@ import (
...
-	ProposerIndex   uint64 `yaml:"proposer_index"`
-	Blob            string `yaml:"blob"`
-	KzgCommitment   string `yaml:"kzg_commitment"`
-	KzgProof        string `yaml:"kzg_proof"`
+	Index                       uint64                          `yaml:"index"`
+	Blob                        string                          `yaml:"blob"`
+	KZGCommitment               string                          `yaml:"kzg_commitment"`
+	KZGProof                    string                          `yaml:"kzg_proof"`
+	SignedBlockHeader           *phase0.SignedBeaconBlockHeader `yaml:"signed_block_header"`
+	KZGCommitmentInclusionProof [17]string                      `yaml:"kzg_commitment_inclusion_proof"`
```
Comment (avalonche) [inline spec/deneb/blobsidecar_yaml.go:33]: why not use `kzgCommitmentProofElements`?

File: spec/deneb/blobsidecar_test.go
Comment (avalonche) [inline spec/deneb/blobsidecar_test.go:1]: some logging and commented out tests?

File: api/versionedproposal.go [hunk 3]
```diff
@@ -509,23 +509,45 @@ func (v *VersionedProposal) Timestamp() (uint64, error) {
...
 }
 
-// BlobSidecars returns the blob sidecars of the proposal.
-func (v *VersionedProposal) BlobSidecars() ([]*deneb.BlobSidecar, error) {
+// Blobs returns the blobs of the proposal.
+func (v *VersionedProposal) Blobs() ([]deneb.Blob, error) {
 	switch v.Version {
 	case spec.DataVersionPhase0:
-		return make([]*deneb.BlobSidecar, 0), nil
+		return make([]deneb.Blob, 0), nil
```
Comment (avalonche) [inline api/versionedproposal.go:516]: this should return an unsupported version error? or at least nil for the array returned

File: api/v1/deneb/blockcontents_test.go
Comment (avalonche) [inline api/v1/deneb/blockcontents_test.go:1]: deleted tests?

File: api/v1/deneb/signedblockcontents_ssz.go [hunk 4]
```diff
@@ -1,160 +0,0 @@
...
-// Version: 0.1.3
-package deneb
-
-import (
-	"github.com/attestantio/go-eth2-client/spec/deneb"
-	ssz "github.com/ferranbt/fastssz"
-)
-
-// MarshalSSZ ssz marshals the SignedBlockContents object
-func (s *SignedBlockContents) MarshalSSZ() ([]byte, error) {
```
Comment (xenowits) [inline api/v1/deneb/signedblockcontents_ssz.go:12]: Why did we delete this file?

File: spec/deneb/blobsidecar_test.go
Comment (mcdee) [reply spec/deneb/blobsidecar_test.go:1]: Thanks; fixed.

File: spec/deneb/blobsidecar_yaml.go (hunk 2 above)
Comment (mcdee) [reply spec/deneb/blobsidecar_yaml.go]: Good point.  Updated.

File: api/versionedproposal.go (hunk 3 above)
Comment (mcdee) [reply api/versionedproposal.go]: Updated to return an unsupported version error.

File: api/v1/deneb/signedblockcontents_ssz.go (hunk 4 above)
Comment (mcdee) [reply api/v1/deneb/signedblockcontents_ssz.go]: It's back now.

File: api/v1/deneb/blockcontents_test.go
Comment (mcdee) [reply api/v1/deneb/blockcontents_test.go:1]: Reinstated.

**Pull request merged**